multi-ai-cli/backends/grok_web.py

```
@@ -0,0 +1,108 @@
+
+        try:
+            # Use impersonate if using curl_cffi for better WAF bypass
+            kwargs = {"headers": headers, "json": payload, "cookies": self.cookies, "stream": True}
+            if hasattr(curl_requests, "Session"):
+                s = curl_requests.Session()
+                if hasattr(s, "impersonate"):
+                    resp = s.post(self.base_url, impersonate="chrome120", **kwargs)
+                else:
+                    resp = s.post(self.base_url, **kwargs)
+            else:
+                resp = curl_requests.post(self.base_url, **kwargs)
```

Ruff (0.16.1)

[error] Probable use of `requests` call without timeout

(S113)